### PR TITLE
Allow not specify twice kafka tables in kafka plugin

### PR DESCRIPTION
**Pull request by satybald**
Fixes: #11294 

Kafka plugin requires to specify kafka names in kafka.table-names and also in schema file, if it is defined. The PR allows not specify a table in the kafka.table-names fields if it's defined in the schema.

### CODE REVIEW

**Comment by satybald** [comment]
Two test cased are failed due to OOM. Not sure what was the cause for that. @electrum @ebyhr could you please help with advice?

```
ERROR] Failures: 
[ERROR]   TestMemoryTracking.testLocalTotalMemoryLimitExceeded:158 expected [Query exceeded local total memory limit of 1GB] but found [Query exceeded per-node total memory limit of 1GB]
[ERROR] com.facebook.presto.operator.TestHashAggregationOperator.testHashBuilderResizeLimit(com.facebook.presto.operator.TestHashAggregationOperator)
[ERROR]   Run 1: TestHashAggregationOperator.testHashBuilderResizeLimit:452 Â» ExceededMemoryLimit
[ERROR]   Run 2: TestHashAggregationOperator.testHashBuilderResizeLimit:452 Â» ExceededMemoryLimit
[INFO] 
[ERROR] com.facebook.presto.operator.TestHashAggregationOperator.testMemoryLimit(com.facebook.presto.operator.TestHashAggregationOperator)
[ERROR]   Run 1: TestHashAggregationOperator.testMemoryLimit:341 Â» ExceededMemoryLimit Query ex...
[ERROR]   Run 2: TestHashAggregationOperator.testMemoryLimit:341 Â» ExceededMemoryLimit Query ex...
[INFO] 
[ERROR] com.facebook.presto.operator.TestHashJoinOperator.testMemoryLimit(com.facebook.presto.operator.TestHashJoinOperator)
[ERROR]   Run 1: TestHashJoinOperator.testMemoryLimit:957->buildHash:1199->buildLookupSource:1297 Â» ExceededMemoryLimit
[ERROR]   Run 2: TestHashJoinOperator.testMemoryLimit:957->buildHash:1199->buildLookupSource:1297 Â» ExceededMemoryLimit
[ERROR]   Run 3: TestHashJoinOperator.testMemoryLimit:957->buildHash:1199->buildLookupSource:1297 Â» ExceededMemoryLimit
[ERROR]   Run 4: TestHashJoinOperator.testMemoryLimit:957->buildHash:1199->buildLookupSource:1297 Â» ExceededMemoryLimit
[INFO] 
[ERROR] com.facebook.presto.operator.TestHashSemiJoinOperator.testMemoryLimit(com.facebook.presto.operator.TestHashSemiJoinOperator)
[ERROR]   Run 1: TestHashSemiJoinOperator.testMemoryLimit:377 Â» ExceededMemoryLimit Query excee...
[ERROR]   Run 2: TestHashSemiJoinOperator.testMemoryLimit:377 Â» ExceededMemoryLimit Query excee...
[INFO] 
[ERROR]   TestOrderByOperator.testMemoryLimit:189 Â» ExceededMemoryLimit Query exceeded p...
[ERROR]   TestWindowOperator.testMemoryLimit:230 Â» ExceededMemoryLimit Query exceeded pe...
[ERROR]   TestMathFunctions.testRound:764->AbstractTestFunctions.assertFunction:94 Â» Semantic
[ERROR] Failures: 
[ERROR]   TestMemoryManager.testQueryMemoryPerNodeLimit:346 Â» Runtime Query exceeded per...
[ERROR]   TestMemoryManager.testQueryTotalMemoryLimit:333 Â» Runtime Query exceeded distr...
[ERROR] com.facebook.presto.memory.TestMemoryManager.testQueryUserMemoryLimit(com.facebook.presto.memory.TestMemoryManager)
[ERROR]   Run 1: TestMemoryManager.testQueryUserMemoryLimit:320 Â» Runtime Query exceeded distri...
[ERROR]   Run 2: TestMemoryManager.testQueryUserMemoryLimit:320 Â» Runtime Query exceeded distri...
[INFO] 
```

File: presto-kafka/src/main/java/com/facebook/presto/kafka/KafkaTableDescriptionSupplier.java [hunk 1]
```diff
@@ -64,7 +65,7 @@
     @Override
     public Map<SchemaTableName, KafkaTopicDescription> get()
     {
-        ImmutableMap.Builder<SchemaTableName, KafkaTopicDescription> builder = ImmutableMap.builder();
+        ImmutableMap.Builder<SchemaTableName, KafkaTopicDescription> tableDefintionBuilder = ImmutableMap.builder();
```
Comment (electrum) [inline presto-kafka/src/main/java/com/facebook/presto/kafka/KafkaTableDescriptionSupplier.java]: Typo: `tableDefintionBuilder` -> `tableDefinitionBuilder `

File: presto-kafka/src/main/java/com/facebook/presto/kafka/KafkaTableDescriptionSupplier.java [hunk 2]
```diff
@@ -73,43 +74,32 @@
...
-                    log.debug("Kafka table %s.%s: %s", schemaName, table.getTableName(), table);
-                    builder.put(new SchemaTableName(schemaName, table.getTableName()), table);
+                    log.debug("Found Kafka Table definition for %s.%s: %s", schemaName, table.getTableName(), table);
+                    tableDefintionBuilder.put(new SchemaTableName(schemaName, table.getTableName()), table);
                 }
             }
-
-            Map<SchemaTableName, KafkaTopicDescription> tableDefinitions = builder.build();
-
+            final Map<SchemaTableName, KafkaTopicDescription> tableDefinitions = tableDefintionBuilder.build();
```
Comment (electrum) [inline presto-kafka/src/main/java/com/facebook/presto/kafka/KafkaTableDescriptionSupplier.java]: Code style nit: we don't declare local variables (or parameters) as `final`

File: presto-kafka/src/main/java/com/facebook/presto/kafka/KafkaTableDescriptionSupplier.java [hunk 3]
```diff
@@ -73,43 +74,32 @@
...
+                        log.debug("Created dummy Table definition for %s", tableName);
+                        builder.put(tableName, new KafkaTopicDescription(
+                                tableName.getTableName(),
+                                Optional.ofNullable(tableName.getSchemaName()),
+                                definedTable,
+                                Optional.of(new KafkaTopicFieldGroup(DummyRowDecoder.NAME, Optional.empty(), ImmutableList.of())),
+                                Optional.of(new KafkaTopicFieldGroup(DummyRowDecoder.NAME, Optional.empty(), ImmutableList.of()))));
+                    });
+
+            //Put defined schema tables
```
Comment (electrum) [inline presto-kafka/src/main/java/com/facebook/presto/kafka/KafkaTableDescriptionSupplier.java]: Maybe rephrase this as
```java
// Add definitions for tables with explicitly defined schemas

File: presto-kafka/src/main/java/com/facebook/presto/kafka/KafkaTableDescriptionSupplier.java [hunk 4]
```diff
@@ -73,43 +74,32 @@
...
-                            Optional.of(new KafkaTopicFieldGroup(DummyRowDecoder.NAME, Optional.empty(), ImmutableList.of())),
-                            Optional.of(new KafkaTopicFieldGroup(DummyRowDecoder.NAME, Optional.empty(), ImmutableList.of()))));
-                }
-            }
-
-            return builder.build();
+            Map<SchemaTableName, KafkaTopicDescription> builder = new HashMap<>();
+
+            // A dummy table definition only supports the internal columns.
+            tableNames.stream()
```
Comment (electrum) [inline presto-kafka/src/main/java/com/facebook/presto/kafka/KafkaTableDescriptionSupplier.java]: No need to call `stream()`. You can call `forEach()` directly on the `Set`.

Code style nit: we generally prefer to use a regular for-each loop to do simple iteration and reserve `forEach` for times when it is required, such as iterating over the result of a stream transformation, or when the body can be a method reference:
```java
values.forEach(this::process);
```
In this case, it seems like a regular for-each would be simpler.

File: presto-kafka/src/main/java/com/facebook/presto/kafka/KafkaTableDescriptionSupplier.java [hunk 5]
```diff
@@ -129,11 +119,16 @@
...
-        List<String> parts = Splitter.on('.').splitToList(schemaTableName);
-        checkArgument(parts.size() == 2, "Invalid schemaTableName: %s", schemaTableName);
-        return new SchemaTableName(parts.get(0), parts.get(1));
+        try {
+            checkArgument(!isNullOrEmpty(schemaTableName), "schemaTableName is null or is empty");
+            List<String> parts = Splitter.on('.').splitToList(schemaTableName);
+            checkArgument(parts.size() == 2, "Invalid schemaTableName: %s", schemaTableName);
+            return new SchemaTableName(parts.get(0), parts.get(1));
+        }
+        catch (IllegalArgumentException iae) {
```
Comment (electrum) [inline presto-kafka/src/main/java/com/facebook/presto/kafka/KafkaTableDescriptionSupplier.java]: Name the exception argument `ignored` so it's clear to the reader and IntelliJ that we are intentionally ignoring it.
